runtime/lua/vim/lsp.lua

```diff
@@ -651,7 +673,18 @@ function lsp.start(config, opts)
 
   if not config.root_dir and opts._root_markers then
     config = vim.deepcopy(config)
-    config.root_dir = vim.fs.root(bufnr, opts._root_markers)
+
+    if type(opts._root_markers) == 'table' then
```

Comment:
Is this condition ever false? I thought `opts._root_markers` is also a table when it's passed below (line 686). Did you mean this instead:
```suggestion
    if type(opts._root_markers[1]) == 'table' then
```

Reply:
Good catch. I don't think this condition is necessary at all.

Reply:
If that's the case I would add a `validate('root_markers', opts._root_markers, 'table')` just before.

Reply:
I pushed a new approach.

Reply:
`root_markers` is not a function, doesn't matter if it works.